### ISSUE TITLE
Change convertors order (Performance reasons)

### ISSUE THREAD

**Issue by Quazz** [yireo/Yireo_Webp2#77]
Currently, cwebp is the first converter tried as per https://github.com/yireo/Yireo_Webp2/blob/c1f494481f3e50b63a48e95b9a5a3c53b061c7c7/Config/Config.php#L57

This then ends up using the rosell-dk/webp-convert dependency. In that library, cwebp is called using the PHP exec function. This is where the performance bottleneck is. Exec is incredibly slow. (slow product saving and category pages having their images load in very late caused me to investigate). As far as I could find online this goes all the way to a slow C function popen used by exec internally. 

Changing the order of convertors (putting gd and imagick before cwebp since they have PHP extensions and don't need exec) fixes the issue assuming you have webp support enabled in those extensions. 

This, however, seems to fail to convert to webp format on save (it is configured in settings of course)

I don't have scientific measuring for you (but I can tell you that a product save where I add 3 images took more than 3 times longer to save than normal), but you can of course test for yourself and compare the difference. 

tl;dr Try php extensions before shell calls for massive performance gains. Look whether conversion on save needs fixing for other convertors.

**Comment by bernieu2** [comment]
Ah, thanks for raising this. I noticed very slow saving of products as well. 45-60s for one image. After changing to gd, that dropped back down to 3-5s. But I haven't experienced any issue with convert on save. I can see that is working fine with gd.

I did test with resize command after clearing image cache. All these products only have 1 image - and there are 8 image sizes per product. 

original + webp = 2464 images
 **with gd:** 
```
 bin/magento catalog:images:resize
 154/154 [============================] 100% 4 mins 90.5 MiB     
```
 
 **with cwebp:**
```
 bin/magento catalog:images:resize
 16/154 [==>-------------------------]  10% 7 mins 90.5 MiB 
```

Obviously, I stopped the cwebp process before finishing.. I think you can extrapolate from there.

**Comment by btekbtek** [comment]
Thanks for this. I had this error:
Warning: imagecreatefrompng(): gd-png:  fatal libpng error.

I wrote Plugin to remove GD from array and it's working now. Maybe  I have old libraries of gd, but for now is enough before Magento update.

**Comment by jissereitsma** [comment]
Thanks @btekbtek @bernieu2 @Quazz for sharing. I think the obvious first step would be to allow you to configure the convertors array - this is why it was already the `Config` class, but I never thought it would actually be needed. Working on it right now.

**Comment by jissereitsma** [comment]
Configuring the convertors is now implemented in version 0.10.8. Does that help you out?

**Comment by jissereitsma** [comment]
Closing this because of inactivity. Most likely the original issue was fixed with releases in between. Feel free to re-open this issue if needed.
